components/discord/actions/common/common.mjs

```diff
@@ -0,0 +1,69 @@
+import discord from "../../discord.app.mjs";
+
+export default {
+  props: {
+    discord,
+    channel: {
+      type: "$.discord.channel",
+      appProp: "discord",
+    },
+    message: {
+      propDefinition: [
+        discord,
+        "message",
+      ],
+    },
+    threadID: {
+      propDefinition: [
+        discord,
+        "threadID",
+      ],
+    },
+    username: {
+      propDefinition: [
+        discord,
+        "username",
+      ],
+    },
+    avatarURL: {
+      propDefinition: [
+        discord,
+        "avatarURL",
+      ],
+    },
+    includeSentViaPipedream: {
+      propDefinition: [
+        discord,
+        "includeSentViaPipedream",
+      ],
+    },
+  },
+  methods: {
+    getUserInputProps(omit = [
+      "discord",
+    ]) {
+      return Object.keys(this)
```

Comment:
Optional:
Just in case you want to restrict the use of `this` in just one plece you can also make use of `Object.entries(this)`